h2o-docs/src/product/data-munging/target-encoding.rst

```diff
@@ -146,136 +216,144 @@ Now we will perform target encoding on ``addr_state`` to see if this representat
+   .. code-block:: r
+
+    train$fold <- h2o.kfold_column(train, nfolds=5, seed = 1234)
+    te_map <- h2o.target_encode_fit(train, x = list("addr_state"), 
+                                    y = response, fold_column = "fold")
 
-.. code:: r
+   .. code-block:: python
 
-    train$fold <- h2o.kfold_column(train, 5, seed = 1234)
-    te_map <- h2o.target_encode_create(train, x = list("addr_state"), 
-                                       y = response, fold_column = "fold")
-    head(te_map$addr_state)
+    from h2o.targetencoder import TargetEncoder
+    fold = train.kfold_column(n_folds=5, seed=1234)
+    predictor = "addr_state"
 
-::
+    te_map = TargetEncoder(x=predictor, y=response, fold_column=fold, seed=1234)
+    te_map.fit(train)
 
-    ##   addr_state fold numerator denominator
-    ## 1         AK    0         3          11
-    ## 2         AK    1         0           5
-    ## 3         AK    2         1          10
-    ## 4         AK    3         2          13
-    ## 5         AK    4         1           7
-    ## 6         AL    0         7          52
+Transform Target Encoding
+'''''''''''''''''''''''''
 
-Apply the target encoding to our training and testing data. For our training data, we will use the parameters:
+Apply the target encoding to our training and testing data. 
 
--  ``holdout_type``: "KFold"
--  ``blended_avg``: TRUE
--  ``noise_level``: NULL (by default it will add 0.01 \* range of y of random noise)
+**Apply Target Encoding to Training Dataset** 
 
-.. code:: r
+For our training data, we will use the parameters:
 
-    ext_train <- h2o.target_encode_apply(train, x = list("addr_state"), y = response, 
-                                         target_encode_map = te_map, holdout_type = "KFold",
-                                         fold_column = "fold",
-                                         blended_avg = TRUE, noise_level = 0, seed = 1234)
+-  ``holdout_type="kfold"``
+-  ``blended_avg=TRUE``
+-  ``noise=0.2``
 
-    head(ext_train[c("addr_state", "fold", "TargetEncode_addr_state")])
+.. example-code::
+   .. code-block:: r
 
-::
+    ext_train <- h2o.target_encode_transform(train, x = list("addr_state"), y = response, 
+                                             target_encode_map = te_map, holdout_type = "kfold",
+                                             fold_column="fold", blended_avg = TRUE, 
+                                             inflection_point=3, smoothing=1, seed = 1234)
 
-    ##   addr_state fold TargetEncode_addr_state
-    ## 1         AK    0               0.1212239
-    ## 2         AK    0               0.1212239
-    ## 3         AK    0               0.1212239
-    ## 4         AK    0               0.1212239
-    ## 5         AK    0               0.1212239
-    ## 6         AK    0               0.1212239
+   .. code-block:: python
 
-For our testing data, we will use the parameters:
+    ext_train = TargetEncoder(x=predictor, y=response, target_encode_map=te_map,
```

Comment:
@angela0xdata please read the tests which I have sent you names of before carefully. `titanic_with_te_kfoldstrategy` test in `/h2o-py/tests/testdir_algos/automl/target_encoding/pyunit_automl_gbm_target_encoding.py` might help you the most.  
Pay attention to parameters that you are passing and to what objects. You should not pass 'target_encode_map=te_map,' to `TargetEncoder` object. Also when you do `transform` you should specify not only frame you are going to apply target encoding to.

Reply:
@deil87 QQ. Why does R use a target_encode_map but Python does not?

Reply:
@deil87 NVM... I see now.

Reply:
@angela0xdata PR has WIP label. Is it ready for review?

Reply:
yes, i've requested a review. I'm still getting errors with my Python examples. I also get errors when I run this python unit test: https://github.com/h2oai/h2o-3/blob/78dbc189c3d6273d8f7b8a61a0199eb5c7c75d5a/h2o-py/tests/testdir_algos/automl/target_encoding/pyunit_automl_gbm_target_encoding.py

Reply:
@angela0xdata tests in `pyunit_automl_gbm_target_encoding.py` should be fine as Mr. Jenkins check them. It might mean that something is wrong with your local setup. Maybe you are using old version of the backend. Can you share errors that you are getting in Slack?

Reply:
@deil87 I was trying to run that test as a user would (so like a demo), not as Jenkins would, so I tried to make some adjustments. But I'm no expert.